### ISSUE TITLE
Python 3.12 warnings

### ISSUE THREAD

**Issue by leahneukirchen** [scottchiefbaker/dool#63]
##### SUMMARY

Running dool on Python 3.12 yields warnings.

##### ISSUE TYPE

 - Bug Report

##### DOOL VERSION

Dool 1.3.0

##### OS / ENVIRONMENT

Linux x86_64

##### STEPS TO REPRODUCE

```
% dool --version
/usr/bin/dool:806: SyntaxWarning: invalid escape sequence '\d'
  self.diskfilter = re.compile('^([hsv]d[a-z]+\d+|cciss/c\d+d\d+p\d+|dm-\d+|md\d+|mmcblk\d+p\d0|VxVM\d+|nvme\d+n\d+p\d+)$')
/usr/bin/dool:1011: SyntaxWarning: invalid escape sequence '\d'
  self.diskfilter = re.compile('^([hsv]d[a-z]+\d+|cciss/c\d+d\d+p\d+|dm-\d+|md\d+|mmcblk\d+p\d0|VxVM\d+|nvme\d+n\d+p\d+)$')
/usr/bin/dool:1159: SyntaxWarning: invalid escape sequence '\d'
  self.totalfilter = re.compile('^(lo|bond\d+|face|.+\.\d+)$')
/usr/bin/dool:2291: SyntaxWarning: invalid escape sequence '\d'
  search = re.compile('^cpu\d+')
/usr/bin/dool:2308: SyntaxWarning: invalid escape sequence '\d'
  m = re.match('ide/host(\d)/bus(\d)/target(\d)/lun(\d)/disc', device)
/usr/bin/dool:2315: SyntaxWarning: invalid escape sequence '\d'
  m = re.match('cciss/(c\dd\d)', device)
Dool 1.3.0
```

##### EXPECTED RESULTS

No warnings.

##### ACTUAL RESULTS

The`\d` should be written as `\\d` I think.  (or use Python raw strings `r'\d'`)

**Comment by scottchiefbaker** [comment]
This should already be fixed on the `next` branch in commit 30168e34f59a8ad40e973ffa45dbe35e7bdd7053. Give it a test and let me know.

**Comment by leahneukirchen** [comment]
Ah, indeed. I forgot `next` is a thing. ;)
